dd-sdk-android/src/test/kotlin/com/datadog/android/core/internal/data/upload/DataUploadRunnableTest.kt

```diff
@@ -207,6 +207,101 @@ internal class DataUploadRunnableTest {
         )
     }
 
+    @ParameterizedTest
+    @EnumSource(
+        SystemInfo.BatteryStatus::class,
+        mode = EnumSource.Mode.INCLUDE
+    )
+    fun `M send batch W device has external power source`(
+        batteryStatus: SystemInfo.BatteryStatus,
+        @Forgery batch: Batch,
+        forge: Forge
+    ) {
+        val systemInfo = SystemInfo(
+            batteryStatus = batteryStatus,
+            batteryLevel = forge.anInt(0, 100),
+            externalPowerSource = true
+        )
+        whenever(mockSystemInfoProvider.getLatestSystemInfo()) doReturn systemInfo
+        whenever(mockReader.lockAndReadNext()) doReturn batch
+        whenever(mockDataUploader.upload(batch.data)) doReturn UploadStatus.SUCCESS
+
+        testedRunnable.run()
+
+        verify(mockReader).drop(batch)
+        verify(mockReader, never()).release(batch)
+        verify(mockDataUploader).upload(batch.data)
+        verify(mockThreadPoolExecutor).schedule(
+            same(testedRunnable),
+            any(),
+            eq(TimeUnit.MILLISECONDS)
+        )
+    }
+
+    @ParameterizedTest
+    @EnumSource(
+        SystemInfo.BatteryStatus::class,
+        names = ["DISCHARGING", "NOT_CHARGING"],
+        mode = EnumSource.Mode.INCLUDE
+    )
+    fun `M send batch W device has external power source { battery low }`(
```

Comment:
Your test use a random battery level between 0 and 100, so it could be 90, which doesn't look like a low battery. Based on the enum source, it seems you're checking for unplugged state ?

Reply:
Same thing as above